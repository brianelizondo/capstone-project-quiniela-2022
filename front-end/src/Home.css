.Home-quiniela-list a{
    text-decoration: none !important;
    color: black;
}

.Quiniela-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 30px;
    margin-right: 12px;
    margin-left: 12px;
}
.Quiniela-card-pos-number{
    font-family: 'Rubik', sans-serif;
    font-size: 65px;
    font-weight: 700;
    color:black;
    display: flex;
    align-items: center;
    text-align: center;
}
.Quiniela-card-logo-team{
    max-width: 80px;
    max-height: 100px;
}
.Quiniela-card-text{
    font-size: 14px;
    display: flex;
    align-items: center;
}
.Quiniela-card-text div{
    width: 100% !important;
}
.Quiniela-card-text span{
    font-weight: 700;
}

.Quiniela-card-points{
    font-family: 'Rubik', sans-serif;
    font-size: 45px;
    font-weight: 700;
    color:black;
}
.Quiniela-card-points-number{
    color: black;
    font-size: 45px;
    font-weight: 700;
    line-height: 40px;
    text-align: center;
    padding-top: 20px;
}
.Quiniela-card-points-text{
    color: gray;
    font-size: 13px;
    font-weight: 400;
    text-align: center;
    text-transform: uppercase;
}
