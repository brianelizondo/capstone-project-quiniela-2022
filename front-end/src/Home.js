import React, { useState, useEffect } from 'react';
import { Row, Col } from 'react-bootstrap';
import Loading from './Loading';
import QuinielaListCard from './QuinielaListCard';
import './Home.css';

// import APIFootball api
import APIFootball from "./api-football";

function Home(){
    // initial state
    const [loading, setLoading] = useState(false);
    const [quinielas, setQuinielas] = useState([]);

    useEffect(() => {
        setLoading(true);
        async function getQuinielas() {
            const resp = await APIFootball.getQuinielas();
            setQuinielas(resp);
        }
        getQuinielas();
        setLoading(false);
    }, []);

    if(loading){
        return <Loading />;
    }

    return (
        <div className="Home col-md-8 offset-md-2">
            <Row>
                <Col>
                    <h1 className='section-title'>Quinielas World Cup 2022</h1>
                </Col>
            </Row>
            <Row>
                <Col>
                    <h5 className='section-subtitle'>List of active and participating quinielas</h5>
                </Col>
            </Row>
            <Row>
                <Col>
                    <div className="Home-quiniela-list">
                        { quinielas.map((quiniela, idx) => (<QuinielaListCard key={quiniela.quinielaID} quiniela={quiniela} position={idx} />))}
                    </div>
                </Col>
            </Row>
        </div>
    );
}

export default Home;