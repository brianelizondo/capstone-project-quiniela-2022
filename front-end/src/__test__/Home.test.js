import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import Home from '../Home';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'Home' component without crashing", () => {
        render(
            <MemoryRouter>
                <Home />
            </MemoryRouter>
        );
        const textFound = screen.getByText("List of active and participating quinielas");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'Home' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <Home />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Additional Test", () => {
    it("works when you click on a quiniela card go to the quiniela details", async () => {
        render(
            <MemoryRouter>
                <Home />
            </MemoryRouter>
        );
        const user = userEvent.setup()

        waitFor(() => { expect(screen.getByText("List of active and participating quinielas")).toBeInTheDocument() });
        waitFor(async () => { await user.click(screen.getAllByTestId("quiniela-")) });
        waitFor(() => { expect(screen.getByText("Champion team of this Quiniela")).toBeInTheDocument() });
    });
});