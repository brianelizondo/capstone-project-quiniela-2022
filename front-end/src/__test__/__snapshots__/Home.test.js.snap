// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'Home' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home col-md-8 offset-md-2"
  >
    <h1>
      Quinielas World Cup 2022
    </h1>
    <p>
      List of active and participating quinielas
    </p>
    <div
      class="Home-quiniela-list"
    />
  </div>
</DocumentFragment>
`;
